cms/server/contest/handlers/contest.py

```diff
@@ -228,6 +228,17 @@ def render_params(self):
             if ret["actual_phase"] == 0:
                 ret["phase"] = 0
 
+                if participation.starting_time is not None:
+                    client_ip = self.request.remote_ip
+                    if participation.starting_ip_addresses:
+                        if client_ip not in participation.starting_ip_addresses:
```

Comment:
This isn't correct, you can have `23.12.123.10` in `123.12.123.101`. I guess it's guess enough for us, but it's not clean. If we don't want to bother with doing it correctly but still fix this we can just store`|ip1|ip2|ip3|` and search for `|ip|` (and perhaps update the display, but it's not required)